src/ExtendedMath.f90

```diff
@@ -165,6 +170,56 @@ ELEMENTAL FUNCTION GreatestCommonDivisor(a1,a2) RESULT(b)
 ENDFUNCTION GreatestCommonDivisor
 !
 !-------------------------------------------------------------------------------
+!> @brief solves for the least common multiple of an array of integers
+!> @param u the array of integers to consider
+!> @returns lcm the least common multiple of the array
+!>
+PURE FUNCTION LeastCommonMultiple_A1(u) RESULT(lcm)
+  INTEGER(SIK),INTENT(IN) :: u(:)
+
+  INTEGER(SIK) :: lcm
+  INTEGER(SIK) :: i,a
+
+  lcm = 0
+  IF(SIZE(u) == 0) RETURN
+  IF(ANY(u == 0)) RETURN
+
+  lcm = abs(u(1))
+  IF(ALL(u == u(1))) THEN
+    RETURN
+  ENDIF
+
+  DO i=2,SIZE(u)
+    a = abs(u(i))
+    !In all likelihood there exists an 'a' and 'b' which will cause overflow
+    lcm = (lcm/GreatestCommonDivisor(lcm,a)) * a
+  ENDDO
+
+ENDFUNCTION LeastCommonMultiple_A1
+!
+!-------------------------------------------------------------------------------
+!> @brief solves for the least common multiple of a pair of integers
+!> @param u the first integer to consider
+!> @param v the second integer to consider
+!> @returns lcm the least common multiple of the array
+!>
+ELEMENTAL FUNCTION LeastCommonMultiple_scalar(u,v) RESULT(lcm)
+  INTEGER(SIK),INTENT(IN) :: u
+  INTEGER(SIK),INTENT(IN) :: v
+
+  INTEGER(SIK) :: lcm
+  INTEGER(SIK) :: a,b
+
+  lcm = 0
+  IF(u == 0 .OR. v == 0) RETURN
+  a = ABS(u)
+  b = ABS(v)
+
+  !In all likelihood there exists an 'a' and 'b' which will cause overflow
```

Comment:
```
a=HUGE(1_SIK)
b=HUGE(1_SIK)-1
```
;)

Reply:
lol, yes that is one. Now just to identify how ever many more and we can guard against it.